### PR TITLE
Feature bar chart

### PR DESCRIPTION
**Pull request by lbmerchant93**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [x] Improves an existing feature
- [x] Fixes a bug
- [x] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
Connect the BarChart to the Course Dashboard. Passes props from the state of course dashboard to call the calculation functions which returns the data to populate the graph depending on the selected activity. Tests all features and pass all tests. 

## *What are the relevant tickets?*
Closes _

## *Bugs:*
Might need to modify the cypress testing when the other components are brought in.
Data for Jest test could be separated out into a different file, just included it in the test until all of the other mock data is together. 

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have run Lighthouse/Wave checks for accessibility
- [x] New and existing unit tests pass locally with my change

### CODE REVIEW

File: cypress/integration/BarChart_spec.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+describe('Dashboard View', () => {
```
Comment (srslie) [inline cypress/integration/BarChart_spec.js:1]: I don't know how to test the canvas either so this looks good!

File: pages/courseDashboard.js [hunk 2]
```diff
@@ -2,36 +2,46 @@ import React, { useContext, useState, useEffect } from 'react'
...
-        setPercentageByActivity(calculations.getPercentages(courseModules.data.activityTotals, 'activity'))
-
-        setSharedState({
-          ...sharedState,
-          [courseId]: courseModules.data.course
-        })
+        if(courseModules) {
+          setCourse(courseModules.data.course)
+          setActivityTotals(courseModules.data.activityTotals)
+          if(courseModules.data.activityTotals !== null) {
```
Comment (srslie) [inline pages/courseDashboard.js:23]: Thanks for bug-fixing these!

**Review by srslie — APPROVED** [review]
Thank you!

File: cypress/integration/BarChart_spec.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  const baseUrl = 'http://localhost:3000/'
```
Comment (camaragon) [inline cypress/integration/BarChart_spec.js:2]: I think this cypress testing of the bar chart looks great! I'm doing something similar with the circle charts

**Review by camaragon — APPROVED** [review]
The bar charts look great!

**Pull request merged**